include/oneapi/dpl/pstl/execution_impl.h

```diff
@@ -111,41 +79,72 @@ template <typename _ExecutionPolicy>
 using __allow_parallel =
     typename __internal::__policy_traits<typename ::std::decay<_ExecutionPolicy>::type>::__allow_parallel;
 
-template <typename _ExecutionPolicy, typename... _IteratorTypes>
-auto
-__is_vectorization_preferred(_ExecutionPolicy& __exec)
-    -> decltype(__internal::__lazy_and(__exec.__allow_vector(),
-                                       typename __internal::__is_random_access_iterator<_IteratorTypes...>::type()))
+//------------------------------------------------------------------------
+// backend selector with tags
+//------------------------------------------------------------------------
+
+struct __tbb_backend {};
+
+template <class _IsVector>
+struct __serial_tag
+{
+    using __is_vector = _IsVector;
+};
+
+template <class _IsVector>
+struct __parallel_tag
+{
+    using __is_vector = _IsVector;
+    // backend tag can be change depending on
+    // TBB availability in the environment
+    using __backend_tag = __tbb_backend;
```

Comment:
1) Agree with "the member field"...
2) Also, it makes to add the option with openMP backend, to show how to a host backend will be switched (by macro)
```
using __backend_tag = 
#ifdef PSTL_TBB_BACKEND
__tbb_backend;
#elif PSTL_OPENMP_BACKEND
__openmp_backend;
#else
__serial_backend;
#endif
```